verifier/src/report_body.rs

```diff
@@ -244,18 +244,55 @@ impl IntoVerificationError for MiscellaneousSelect {
 }
 
 /// Verifier for ensuring [`MrEnclave`] values are equivalent.
+///
+/// The Intel SDK docs refer to this as "Strict Enclave Modification Policy"
 pub type MrEnclaveVerifier = EqualityVerifier<MrEnclave>;
 impl IntoVerificationError for MrEnclave {
     fn into_verification_error(expected: Self, actual: Self) -> VerificationError {
         VerificationError::MrEnclaveMismatch { expected, actual }
     }
 }
 
-/// Verifier for ensuring [`MrSigner`] values are equivalent.
-pub type MrSignerVerifier = EqualityVerifier<MrSigner>;
+/// Verifier for ensuring all of the MRSIGNER inputs are sufficient.
+///
+/// The Intel SDK docs refer to this as "Security Enclave Modification Policy"
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct MrSignerVerifier {
+    mr_signer: MrSignerKeyVerifier,
+    product_id: IsvProductIdVerifier,
+    isv_svn: IsvSvnVerifier,
+}
+
+impl MrSignerVerifier {
+    /// Create a new [`MrSignerVerifier`]
+    pub fn new(mr_signer: MrSigner, product_id: IsvProductId, isv_svn: IsvSvn) -> Self {
+        Self {
+            mr_signer: MrSignerKeyVerifier::new(mr_signer),
+            product_id: IsvProductIdVerifier::new(product_id),
+            isv_svn: IsvSvnVerifier::new(isv_svn),
+        }
+    }
+}
+
+impl<E: Accessor<MrSigner> + Accessor<IsvProductId> + Accessor<IsvSvn>> Verifier<E>
+    for MrSignerVerifier
+{
+    type Error = VerificationError;
+    fn verify(&self, evidence: &E) -> CtOption<Self::Error> {
+        let mr_signer_key = self.mr_signer.verify(evidence);
+        let product_id = self.product_id.verify(evidence);
+        let isv_svn = self.isv_svn.verify(evidence);
+
+        let is_some = mr_signer_key.is_some() | product_id.is_some() | isv_svn.is_some();
+
+        CtOption::new(VerificationError::General, is_some)
```

Comment:
This general error is here for now, until #31 gets hashed out.
I've added a task item to #8 to prevent forgetting about this.